butterfly/CoreLayer/Core.py

```diff
@@ -43,6 +42,17 @@ def make_tile_query(self, query, t_index):
         tile_crop = query.all_in_some(t_index)
         return TileQuery(query, t_index, tile_crop)
 
+    def load_cutout(self, query):
+        if not self._cache.get_source(query):
```

Comment:
If the source isn't in the cache, we call make a dummy `TileQuery` at the first index [0,0,0] to get information about the whole datasource. Then we pass that information back to the `DataQuery`. This is a bit silly, but it's necessary to keep it such that `DataQuery` only ever interacts with any `ImageLayer` class through the `TileQuery`.